### PR TITLE
add .inner_html() builder method

### PR DESCRIPTION
**Pull request by alsuren**
This is useful for injecting SVGs into the dom.

I have not tested this with SSR or hydration, because I'm not using them in my current project.

Screenshot 2022-03-07 at 11 56 21

### CODE REVIEW

**Comment by alsuren** [comment]
I guess this relates to https://github.com/sycamore-rs/sycamore/issues/288 - we have tried a few things in 0.7-land, like:
* building svgs by hand using the builder api (typically a bad idea, because you end up with an svg that it impossible to view or edit outside of rust-land) 
* falling back to the view!{} macro (okay, but typically forces you to add another layer of div or span that you don't really want).
* encoding the svg as a data url:

	let icon = format!(
		"data:image/svg+xml;charset=UTF-8,{}",
		utf8_percent_encode(source.icon(), ASCII_SET)
	);
        img().attr("src", icon).build()

Since this patch is tiny, we might backport it to 0.7 as well, so I can see how it feels when used with our existing codebase. That way, we give an experience report.

File: packages/sycamore/src/builder.rs [hunk 1]
```diff
@@ -143,6 +143,29 @@ impl<'a, G: GenericNode, F: FnOnce(ScopeRef<'a>) -> G + 'a> ElementBuilder<'a, G
...
+    /// ```compile_fail
+    /// # // This example only fails because we've not supplied a "button_icon.svg" anywhere.
+    /// # // It might be possible to extract some of the SVG constants out of the website dir, and use that here.
+    /// # use sycamore::builder::prelude::*;
+    /// # use sycamore::prelude::*;
+    /// # fn _test<G: GenericNode>(ctx: ScopeRef) -> View<G> {
+    /// h(button).inner_html(include_str!("button_icon.svg"))
+    /// # .view(ctx) }
+    /// ```
+    pub fn inner_html(
```
Comment (alsuren) [inline packages/sycamore/src/builder.rs:162]: There's a whole bunch of potential for strange behaviour lurking in here. I can give it a scarier name if you prefer ;-)

Comment (lukechu10) [reply packages/sycamore/src/builder.rs:162]: I think it should be `dangerously_set_inner_html` to be consistent with the `view!` macro. Not sure if you saw #375 but this PR implements it.

Comment (lukechu10) [reply packages/sycamore/src/builder.rs:162]: Also it would be somewhat hard to backport to v0.7 because the builder API was completed refactored.

**Comment by lukechu10** [comment]
Sorry haven't taken a look in this in a while. It's that and also I'd like to keep the example the way it is right now to be consistent with the non-builder equivalent. I can push a commit on my side to fix that and merge this soon.

**Pull request merged**